main_sweeps.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import json
+import os
+import shutil
+from typing import Any, Dict, Type
+
+import numpy as np
+import pandas as pd
+from pyntcloud import PyntCloud
+from pyquaternion import Quaternion
+
+from argoverse.utils.se3 import SE3
+from argoverse.utils.transform import quat2rotmat
+from main import get_calibration_info, round_to_micros, extract_pc
+from nuscenes.nuscenes import NuScenes
+
+"""
+Converts the nuScenes dataset into the Argoverse format.
+
+In the nuScenes dataset, samples contain annotated data sampled at a frequency
+of 2Hz, whereas sweeps contains all the unannotated data. The capture frequency
+for camera is 12Hz and for lidar it 20Hz. Lidar data is provided in lidar sensor
+coordinate system, annotations are provided in the global city coordinate frame,
+and images are undistorted.
+
+See paper:
+https://arxiv.org/pdf/1903.11027.pdf
+"""
+
+# nuscenes sensor on the left, corresponding argoverse sensor on right
+SENSOR_NAMES = {
+    'LIDAR_TOP': 'lidar',
+    'CAM_FRONT': 'ring_front_center',
+    'CAM_FRONT_LEFT': 'ring_front_left',
+    'CAM_FRONT_RIGHT': 'ring_front_right',
+    'CAM_BACK_LEFT': 'ring_side_left',
+    'CAM_BACK_RIGHT': 'ring_side_right',
+}
+
+CITY_TO_ID = {
+    "singapore-onenorth": "SON",
+    "boston-seaport": "BSP",
+    "singapore-queenstown": "SQT",
+    "singapore-hollandvillage": "SHV",
+}
+
+
+def main(args: argparse.Namespace) -> None:
```

Comment:
do you mind adding a brief explanation of the steps of the conversion algorithm in the docstring?

Reply:
Done